### ISSUE TITLE
consider-using-generator / R1728 false positive for "AsyncGenerator" valid code

### ISSUE THREAD

**Issue by simon-liebehenschel** [pylint-dev/pylint#7271]
### Bug description

Some simple valid Python code for demonstration purposes:
```python
import asyncio

async def main():
    async def my_gen():
        for _ in range(10):
            yield _

    result = tuple([_ async for _ in my_gen()])
    print(result)

asyncio.run(main())
```

Pylint asks me to write 
```python
    result = tuple(_ async for _ in my_gen())
```
which is wrong because you will get an error `TypeError: 'async_generator' object is not iterable`

### Command used

```shell
pre-commit run --all-files
```

### Pylint output

```shell
R1728: Consider using a generator instead 'tuple(artist async for artist in artists)' (consider-using-generator)
```

### Expected behavior

Pylint must understand that `AsyncGenerator` is used here and pylint must show no warnings.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]
```

**Comment by RomanValov** [comment]
Similar issue when using `any`/`all` over async comprehensions:

```python
import asyncio
from collections.abc import AsyncIterator


async def seq() -> AsyncIterator[int]:
    for value in range(10):
        await asyncio.sleep(0.0)
        yield value


async def main() -> None:
    print(any([_ async for _ in seq()]))
    print(all([_ async for _ in seq()]))


if __name__ == "__main__":
    asyncio.run(main())
```

```
R1729: Use a generator instead 'any(_ async for _ in seq())' (use-a-generator)
R1729: Use a generator instead 'all(_ async for _ in seq())' (use-a-generator)
```

**Comment by Pierre-Sassoulas** [comment]
Some await cases are also problematic but not all. 

```python
  async def main():
      print(any([await get(n) for n in [0, 1]]))
```

``any(await get(n) for n in [0, 1])`` raises a TypeError: 'async_generator' object is not iterable. But inner await should still be flagged.

```python
  sum([n for n in await get([1, 2])])  # still a true positive
```

I'm going to add this to your PR, thank you for working on this issue that went under the radar.
